game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/FiringGroupTest.java

```diff
@@ -0,0 +1,187 @@
+package games.strategy.triplea.delegate.battle.steps.fire;
+
+import static games.strategy.triplea.Constants.UNIT_ATTACHMENT_NAME;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.attachments.UnitAttachment;
+import java.util.Comparator;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class FiringGroupTest {
+
+  @Mock GameData gameData;
+  @Mock GamePlayer player;
+
+  @Test
+  public void onlyOneGroupIfNoSuicideOnHit() {
+    final List<Unit> units =
+        List.of(
+            givenUnitWithSuicideOnHit("type1", false), givenUnitWithSuicideOnHit("type2", false));
+    final List<FiringGroup> groups =
+        FiringGroup.groupBySuicideOnHit("test", units, List.of(mock(Unit.class)));
+    assertThat("There should only be one group", groups, hasSize(1));
```

Comment:
nit, the assertion `hasSize(1)` already says there should be one group, what is not said though is why?